design/doc.go

```diff
@@ -2,3 +2,60 @@
 // using goa framework. This generated API accepts HTTP GET/POST/PATCH/PUT/DELETE requests
 // from multiple clients.
 package design
+
+import "strings"
+
+type fieldDesc struct {
+	desc         string
+	mandOnCreate bool
+	mandOnUpdate bool
+}
+
+// desc returns a field description and by default the field is marked as
+// optional in creation and update of resource.
+func desc(str ...string) fieldDesc {
+	var tmp string
+	if len(str) > 0 {
+		tmp = str[0]
+	}
+	return fieldDesc{
+		desc: tmp,
+	}
+}
+
+func (d fieldDesc) String() string {
+	s := d.desc
+	if d.mandOnCreate {
+		if !strings.HasSuffix(strings.TrimSpace(s), ".") {
+			s += ". "
+		}
+		s += "\n This is MANDATORY on creation of resource."
+	}
+	if d.mandOnUpdate {
+		if !strings.HasSuffix(strings.TrimSpace(s), ".") {
+			s += ". "
+		}
+		s += "\n This is MANDATORY on update of resource."
+	}
+	return s
+}
+
+func (d fieldDesc) mandatoryOnCreate(b ...bool) fieldDesc {
```

Comment:
I think the API would be better without any arguments. Let the default value be `false`.

Reply:
No longer applies.